File: coyote/models/rate_limiter_fixed_window_config.py
# this file is @generated
from pydantic import Field

from ..internal.base_model import BaseModel


class RateLimiterFixedWindowConfig(BaseModel):
    max_requests: int = Field(alias="max_requests")
    """Maximum number of requests allowed within the window"""

    window_size: int = Field(alias="window_size")
    """Window size in seconds"""

File: src/models/rate_limiter_fixed_window_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct RateLimiterFixedWindowConfig {
    /// Window size in seconds
    pub window_size: u64,

    /// Maximum number of requests allowed within the window
    pub max_requests: u64,
}

impl RateLimiterFixedWindowConfig {
    pub fn new(window_size: u64, max_requests: u64) -> Self {
        Self {
            window_size,
            max_requests,
        }
    }
}
